src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -2573,11 +2533,23 @@ private bool Fix(int iParam, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
             return _getIsNullableOpt?.Invoke(expr);
         }
 
-        internal static TypeSymbol Merge(TypeSymbol first, TypeSymbol second, AssemblySymbol corLibrary)
+        internal static bool TryMerge(TypeSymbol first, TypeSymbol second, ConversionsBase conversions, out TypeSymbol merged)
+        {
+            TypeSymbolWithAnnotations resultWithAnnotations;
+            bool result = TryMerge(TypeSymbolWithAnnotations.Create(first), TypeSymbolWithAnnotations.Create(second), VarianceKind.None, conversions, out resultWithAnnotations);
+            merged = resultWithAnnotations.TypeSymbol;
+            return result;
```

Comment:
Nit: consider naming something like `success` instead of `result`. Alternatively, consider using tuples for the method return.

Reply:
Thanks. Switched to tuples.

---
In reply to: [223491010](https://github.com/dotnet/roslyn/pull/29727#discussion_r223491010) [](ancestors = 223491010)